modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/SearchAsYouTypeFieldMapper.java

```diff
@@ -692,10 +702,12 @@ protected void doMerge(Mapper mergeWith) {
         super.doMerge(mergeWith);
         SearchAsYouTypeFieldMapper mw = (SearchAsYouTypeFieldMapper) mergeWith;
         if (mw.maxShingleSize != maxShingleSize) {
-            throw new IllegalArgumentException("mapper [" + name() + "] has different maxShingleSize setting, current ["
+            throw new IllegalArgumentException("mapper [" + name() + "] has different [max_shingle_size] setting, current ["
                 + this.maxShingleSize + "], merged [" + mw.maxShingleSize + "]");
         }
-        this.prefixField = (PrefixFieldMapper) this.prefixField.merge(mw);
+        if (prefixField.equals(mw.prefixField) == false) {
+            this.prefixField = (PrefixFieldMapper) this.prefixField.merge(mw.prefixField);
+        }
```

Comment:
Should we use this same logic for the shingle fields below (i.e. only merge if they're unequal)?

Reply:
it's a leftover, we should always merge, I removed the test